src/journal/FutureImpl.cc

```diff
@@ -61,13 +61,13 @@ void FutureImpl::flush(Context *on_safe) {
 }
 
 FutureImplPtr FutureImpl::prepare_flush(FlushHandlers *flush_handlers) {
-  Mutex::Locker locker(m_lock);
+  std::lock_guard locker{m_lock};
   return prepare_flush(flush_handlers, m_lock);
 }
 
 FutureImplPtr FutureImpl::prepare_flush(FlushHandlers *flush_handlers,
-                                        Mutex &lock) {
-  ceph_assert(m_lock.is_locked());
+                                        std::mutex &lock) {
+  // ceph_assert(ceph_mutex_is_locked(m_lock));
```

Comment:
Why does this need to be commented out?

Reply:
because `std::mutex` does not support `is_locked()`. i am commenting it out as a comment to explain that the lock should be acquired at this moment.

@dillaman  if we believe it's important of have this check. i will update `ceph::mutex` or add a dedicated locker type to disable `lockdep`. what do you think?

Reply:
... but why the difference between using `std::mutex` and `ceph::mutex`? Can't it always just be the `ceph::mutex` wrapper? Is it just because `ceph::mutex` will *always* enable lockdep?

Reply:
@dillaman yeah, the idea is to avoid the overhead in the release build, and only enable lockdeps and store the backtrace in the debug build (or the "sanitize" build).

Reply:
@dillaman i added a "customized" `ceph::mutex` for fulfilling this requirement. and the commented out `ceph_assert(ceph_mutex_is_locked(m_lock));` statements are preserved .